test/org/javarosa/model/xform/SMSSerializingVisitorTest.java

```diff
@@ -0,0 +1,161 @@
+package org.javarosa.model.xform;
+
+import org.javarosa.core.PathConst;
+import org.javarosa.core.model.instance.FormInstance;
+import org.javarosa.core.services.transport.payload.ByteArrayPayload;
+import org.javarosa.core.test.FormParseInit;
+import org.javarosa.form.api.FormEntryController;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import static org.javarosa.test.utils.Utils.convertToString;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertNotNull;
+
+/**
+ * Performs a comprehensive suite of tests on the results of a form
+ * being serialized to the SMS format.
+ */
+public class SMSSerializingVisitorTest {
+    private FormParseInit formParser;
+    private String SMS;
+    private static final String delimiter = " ";
+    private FormEntryController formEntryController;
+    private FormInstance formInstance;
+
+    @Before
+    public void setup() throws IOException {
+        formParser = new FormParseInit();
+        formParser.setFormToParse(new File(PathConst.getTestResourcePath(), "sms_form.xml").toString());
+        formEntryController = formParser.getFormEntryController();
+        formInstance = formEntryController.getModel().getForm().getInstance();
+
+        SMSSerializingVisitor serializer = new SMSSerializingVisitor();
+
+        ByteArrayPayload payload = (ByteArrayPayload) serializer.createSerializedPayload(formInstance);
+
+        SMS = convertToString(payload.getPayloadStream());
+        SMS = SMS.replace("\\", "").replace("\\\\", "\\");
```

Comment:
You can replace the two lines with
```
sms = new String(payload.getPayloadBytes(), "UTF-8").replace("\\", "").replace("\\\\", "\\");
```
See my commits here for details: https://github.com/dcbriccetti/javarosa/commits/my-jd-sms
Feel free to cherry pick.